packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,329 @@
+  border-right: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  width: 100%;
+
+  &[data-hovered='true'] {
+    border-right: 1px solid ${({ theme }) => theme.colors.primary.border};
+  }
+
+  height: 100%;
+`
+
+const CustomSelectInput = styled(SelectInputV2)<{
+  width?: number
+}>`
+  ${({ width }) => width && `width: ${width}px;`}
+  &:active {
+    outline: none;
+    box-shadow: none;
+  }
+  &:hover {
+    & > ${StyledInputWrapper} {
+      background: red;
+      border-right-color: ${({ theme }) => theme.colors.primary.border};
+    }
+  }
+`
+
+const UnitInputWrapper = styled(Stack)<{
+  'data-size': 'small' | 'medium' | 'large'
+  'data-focus': boolean
+  'data-success': boolean
+  'data-error': boolean
+  'data-disabled': boolean
+  'data-readonly': boolean
+}>`
+  border: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &:hover,
+  &:focus {
+    text-decoration: none;
+    border-color: ${({ theme }) => theme.colors.primary.border};
+    box-shadow: none;
+  }
+
+  &[data-readonly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+    cursor: default;
+  }
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+    cursor: not-allowed;
+  }
+
+  &[data-size='small'] {
+    height: ${INPUT_SIZE_HEIGHT['small']}px;
+  }
+  &[data-size='medium'] {
+    height: ${INPUT_SIZE_HEIGHT['medium']}px;
+  }
+  &[data-size='large'] {
+    height: ${INPUT_SIZE_HEIGHT['large']}px;
+  }
+  &[data-focus='true'] {
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.primary.border};
+    }
+  }
+  &[data-success='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.success.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.success.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+    & > ${StyledInputWrapper}:hover {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+`
+
+type UnitInputValue = { inputValue: number; unit: string }
+
+type UnitInputProps = {
+  id?: string
+  className?: string
+  name: string
+  disabled?: boolean
+  maxValue?: number
+  minValue?: number
```

Comment:
You can name it `max` and `min` in order to follow same naming as `NumberInputV2`